File: go/283.go
func moveZeroes(nums []int) {
	i := 0
	for _, v := range nums {
		if v != 0 {
			nums[i] = v
			i++
		}
	}
	for i < len(nums) {
		nums[i] = 0
		i++
	}
}

File: java/283.java
class Solution {
    public void moveZeroes(int[] nums) {
        int i = 0;
        int N = nums.length;
        int j = 0;
        for(;i < N;i++){
            if(nums[i] != 0){
                nums[j++] = nums[i];
            }
        }
        for(;j<N;j++){
            nums[j] = 0;
        }
    }
}